src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2737,6 +2738,7 @@ kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
           if (!ostree_kernel_args_delete (kargs, arg, error))
             return FALSE;
         }
+      changed = TRUE;
```

Comment:
I don't think it's possible anyway in this case, but for reading the code it's easier if we just put this in the for-loop body itself so we don't have to think about the case where the array has no element (and ideally we'd just drop the outer `if-statement`s entirely as I think we discussed in the previous PR).

Reply:
Yes, I remember, we can change that now. I did it thinking it was weird to set it to TRUE several times in a loop.

Reply:
Yeah, understood. I guess it comes down a bit to personal preference. I like that having it inside the loop makes the `changed` logic really obvious to see it's right.